### ISSUE TITLE
Ballistica 1.7: V2 Accounts and Workspaces

### ISSUE THREAD

**Issue by Freaku17** [efroemling/ballistica#354]
When will 1.7 release?

**Comment by SnobbishContainer9** [comment]
Eric working on 1.6.8 or newer release, he can't work like robot, please respect him and him work

**Comment by FreakOPP** [comment]
I'm just asking dude ...

**Comment by efroemling** [comment]
No worries; I know I can go for long stretches without updates sometimes so I don't mind being asked.

Anyway, I'm currently towards the end of 1.7 development, and plan to have that released within the next few weeks. I modified the [https://github.com/efroemling/ballistica/wiki/Roadmap](roadmap) a bit; the main feature of 1.7 is going to be V2 accounts and cloud based modding (workspaces). I'll then do a 1.8 release with the asset-packages stuff that was originally going to be part of 1.7.

Most of my server-side work is done. If you'd like, you can go over to https://tools.ballistica.net and create a v2 account and mess around with the workspace editor. It's also a good opportunity to reserve an account tag before all the good ones get taken 🤫.

I'll start releasing 1.7 test builds soon and then those accounts & workspaces will be able to be used in the game. You'll also be able to link V2 accounts to your existing ones so there's no harm in creating a 'new' one now.

Let me know if you have any questions/thoughts about this stuff or if there's anything I can clarify.

**Comment by shayanshokry** [comment]
what will happens to our previous accounts?

**Comment by Freaku17** [comment]
What is 'Asset Packages' on the website? Is it related to the Dependency Component?

**Comment by Dliwk** [comment]
> What is 'Asset Packages' on the website? Is it related to the Dependency Component?

Yes?.. I expect some tool where modders can upload pngs/models/audio files which will be processed to platform-specific files (like .ktx android textures); after all this you'll have to get ready-to-use asset-package@1.0.0. Something like this :man_shrugging:

**Comment by efroemling** [comment]
> > What is 'Asset Packages' on the website? Is it related to the Dependency Component?
> 
> Yes?.. I expect some tool where modders can upload pngs/models/audio files which will be processed to platform-specific files (like .ktx android textures); after all this you'll have to get ready-to-use asset-package@1.0.0. Something like this 🤷‍♂️

Yes, exactly. Asset-Packages and the new dependency system that uses them will be the main feature of 1.8.
Similarly to how 1.7's Workspaces will allow writing code for the game without needing to manually install anything, 1.8's Asset Packages will allow creating assets for the game without needing to manually install anything; everyone who joins a game that makes use of Asset-Packages will be able to automatically download the built assets in the right format for the running platform. I've already got the cloud-based build system working; I just need to get it wired into the game and expose a web-based UI for it. But first finishing 1.7...

**Comment by NishchayKQ** [comment]
> And at some point (though not yet for the initial 1.7 release) I'll upgrade login types such as Google Play Games to be 'V2-native' 
so if you currently log in that way you won't have to do anything to gain V2 features (while still keeping your old v1 data).

Hey so will it auto generate a v2 account for our v1 accounts? Like I thought the game would ask us to create a v2 account and transfer data from v1(play games) to v2 , will this be possible?
I already have a v2 account and like the name it has , while it won't be ideal if our v1 account name gets used for v2. So maybe a choice for us ? 
Thanks

**Comment by NishchayKQ** [comment]
That sounds great ! I will probably wait for v2 native as my play games and new v2 have he same email

But more importantly kinda attached to my old v2 account :)

**Comment by efroemling** [comment]
Sounds good. 👍. I don't think it should take me *too* long to get Google Play logins updated for V2. I just wanted to delay it until a minor 1.7.x point update so I can stress test the 1.7 / V2 stuff with a smaller number of accounts first (there'll be a *lot* of V2 account data flying around once I upgrade Google Play accounts 😅).

**Comment by MatiasMGz** [comment]
So Asset Packages would be like in Minecraft, right?
I am worried about the possible creation of a marketplace where people buy accounts with certain things like Gold Pass, unique names, etc.
It has happened to other games.

**Comment by EraOSBeta** [comment]
> So Asset Packages would be like in Minecraft, right?

They will be somewhat similar.

> I am worried about the possible creation of a marketplace where people buy accounts with certain things like Gold Pass, unique names, etc.
> It has happened to other games.

Who cares? This really isn't an issue as long as there's enough guardrails against hackers and scammers to prevent valuable accounts from being hijacked.

**Comment by MatiasMGz** [comment]
Yeah, i don't think so
